modules/go-sequencer-client/client/client.go

```diff
@@ -24,27 +25,92 @@ type BalanceResponse struct {
 
 // Client is an HTTP tendermint client.
 type Client struct {
-	client *http.HTTP
+	websocket string
+	client    *http.HTTP
 }
 
 func NewClient(url string) (*Client, error) {
+	// Compile the regular expression
+	re := regexp.MustCompile(`^[^:]+://`)
+
 	c, err := http.New(url, "")
 	if err != nil {
 		return nil, err
 	}
 
+	// Replace and print results
+	websocket := re.ReplaceAllString(url, "")
+	websocket = "tcp://" + websocket
 	return &Client{
-		client: c,
+		websocket: websocket,
+		client:    c,
 	}, nil
 }
 
+// BroadcastTx broadcasts a transaction. If async is true, the function will
+// return immediately. The response seen is the generated data used for
+// submitting the transaction. It does not confirm that the data has been
+// included on chain. If async is false, the function will wait for the
+// transaction to be seen on the network.
+func (c *Client) BroadcastTx(ctx context.Context, tx *txproto.SignedTransaction, async bool) (*coretypes.ResultBroadcastTx, error) {
+	if async {
+		return c.BroadcastTxAsync(ctx, tx)
+	}
+	return c.BroadcastTxSync(ctx, tx)
+}
+
+// BroadcastTxAsync broadcasts a transaction and returns immediately.
+func (c *Client) BroadcastTxAsync(ctx context.Context, tx *txproto.SignedTransaction) (*coretypes.ResultBroadcastTx, error) {
+	bytes, err := proto.Marshal(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	return c.client.BroadcastTxAsync(ctx, bytes)
+}
+
+// BroadcastTxSync broadcasts a transaction and waits for the response that
+// confirms the transaction was included.
 func (c *Client) BroadcastTxSync(ctx context.Context, tx *txproto.SignedTransaction) (*coretypes.ResultBroadcastTx, error) {
 	bytes, err := proto.Marshal(tx)
 	if err != nil {
 		return nil, err
 	}
+	result, resultErr := c.client.BroadcastTxSync(ctx, bytes)
```

Comment:
For posterity (discussed in person):
`resultErr` can be `nil` while `result` contains an error message.